core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -98,10 +92,10 @@ class HIPSpace {
                   const size_t arg_logical_size = 0) const;
 
  private:
-  void* impl_allocate(const char* arg_label, const size_t arg_alloc_size,
-                      const size_t arg_logical_size = 0,
-                      const Kokkos::Tools::SpaceHandle =
-                          Kokkos::Tools::make_space_handle(name())) const;
+  void* impl_allocate(const hipStream_t stream, const char* arg_label,
```

Comment:
Just noting that most of the methods here still take `HIP`, not `hipStream_t`.

Also noting that for the `Cuda` backend, it also takes `Cuda` instead of the `cudaStream_t`.

Reply:
I have a similar opinion as @romintomasetti here.

Not just for consistency, but if multi-gpu support is also to be enabled for `HIP` backend in the future, this function will either need to take in a `HIP` instance to extract a device id or add another argument for it.

Reply:
@romintomasetti You are wrong. This is the same interface as the CUDA interface. Look at `impl_allocate_common`.

Reply:
@romintomasetti @ldh4 I have a PR which we can merge after this one https://github.com/kokkos/kokkos/pull/7130 that will add the necessary exe_space impls for multiGPU, no need to have them here.